lib/preview-web/src/PreviewWeb.tsx

```diff
@@ -605,8 +605,20 @@ export class PreviewWeb<TFramework extends AnyFramework> {
 
         if (forceRemount && playFunction) {
           this.disableKeyListeners = true;
-          await runPhase('playing', () => playFunction(renderContext.storyContext));
-          await runPhase('played');
+          await runPhase('playing', async () => {
+            try {
+              await playFunction(renderContext.storyContext);
+              await runPhase('played');
+            } catch (error) {
+              if (error !== IGNORED_EXCEPTION) {
+                logger.error(error);
+                await runPhase('errored', () => {
+                  const { name, message, stack } = error;
+                  this.channel.emit(Events.STORY_THREW_EXCEPTION, { name, message, stack });
+                });
+              }
+            }
+          });
```

Comment:
Can we change this code to not be so nested?
```suggestion
          try {
            await runPhase('playing', async () => playFunction(renderContext.storyContext));    
            await runPhase('played');
          } catch (error) {
            if (error !== IGNORED_EXCEPTION) {
              logger.error(error);
              await runPhase('errored', () => {
                const { name, message, stack } = error;
                this.channel.emit(Events.STORY_THREW_EXCEPTION, { name, message, stack });
              });
            }
          }
```

This would change the semantics however -- if the `playFunction` throws, it would ignore a subsequent abort signal. However I wonder if that is what we want it to do anyway? Should it end up in `aborted` or `errored` in that case?

Reply:
👍